netdata-installer.sh

```diff
@@ -958,39 +958,58 @@ To start netdata run:
-			echo >&2 "Cannot figure out the cron directory to install netdata-updater"
-		else
-			if [ -f "${crondir}/netdata-updater.sh" ]; then
-				progress "Removing incorrect netdata-updater filename in cron"
-				rm -f "${crondir}/netdata-updater.sh"
-			fi
-			progress "Installing new netdata-updater in cron"
+if [ -f "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" ]; then
+	sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" > "${NETDATA_PREFIX}/usr/libexec/netdata-updater" || exit 1
+else
+	sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${NETDATA_SOURCE_DIR}/packaging/installer/netdata-updater.sh" > "${NETDATA_PREFIX}/usr/libexec/netdata-updater" || exit 1
+fi
 
-			rm -f "${crondir}/netdata-updater"
-			if [ -f "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" ]; then
-				sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" > "${crondir}/netdata-updater" || exit 1
-			else
-				sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${NETDATA_SOURCE_DIR}/packaging/installer/netdata-updater.sh" > "${crondir}/netdata-updater" || exit 1
-			fi
+chmod 0755 ${NETDATA_PREFIX}/usr/libexec/netdata-updater
+echo >&2 "Update script is located at ${TPUT_GREEN}${TPUT_BOLD}${NETDATA_PREFIX}/usr/libexec/netdata-updater${TPUT_RESET}"
+echo >&2
 
-			chmod 0755 ${crondir}/netdata-updater
-			echo >&2 "Update script is located at ${TPUT_GREEN}${TPUT_BOLD}${crondir}/netdata-updater${TPUT_RESET}"
-			echo >&2
-			echo >&2 "${TPUT_DIM}${TPUT_BOLD}netdata-updater${TPUT_RESET}${TPUT_DIM} works from cron. It will trigger an email from cron"
-			echo >&2 "only if it fails (it should not print anything when it can update netdata).${TPUT_RESET}"
+# Figure out the cron directory for the distro
+crondir=
+[ -d "/etc/periodic/daily" ] && crondir="/etc/periodic/daily"
+[ -d "/etc/cron.daily" ] && crondir="/etc/cron.daily"
+
+if [ -z "${crondir}" ]; then
+	echo >&2 "Cannot figure out the cron directory to handle netdata-updater activation/deactivation"
+elif [ "${UID}" -ne "0" ]; then
+	# We cant touch cron if we are not running as root
+	echo >&2 "You need to run the installer as root for auto-updating via cron."
+else
+	progress "Check if we must netdata-updater"
+	if [ "${AUTOUPDATE}" = "1" ]; then
+		if [ -f "${crondir}/netdata-updater.sh" ]; then
+			progress "Removing incorrect netdata-updater filename in cron"
+			rm -f "${crondir}/netdata-updater.sh"
+		fi
+
+		echo >&2 "Adding to cron"
+
+		rm -f "${crondir}/netdata-updater"
+		ln -sf "${NETDATA_PREFIX}/usr/libexec/netdata-updater" "${crondir}/netdata-updater"
+
+		echo >&2 "Auto-updating has been enabled. Updater script linked to: ${TPUT_RED}${TPUT_BOLD}${crondir}/netdata-update${TPUT_RESET}"
```

Comment:
netdata-updater

Reply:
Good point, adjusting

Reply:
I still see netdata-update in the message